odoo/addons/base/models/res_company.py

```diff
@@ -353,6 +360,8 @@ def write(self, values):
                 ])
                 for fname in sorted(changed):
                     branches[fname] = company[fname]
+            if company in companies_needs_l10n and company.uninstalled_l10n_module_ids:
+                company.install_l10n_modules()
```

Comment:
Why not batch the installation?

```suggestion
        companies_needs_l10n = companies_needs_l10n.filtered('uninstalled_l10n_module_ids')
        if companies_needs_l10n:
            companies_needs_l10n.install_l10n_modules()
```

Reply:
Tested it, doesn't works well when company is still not yet saved

Reply:
But... it is saved. `super().write` has been called.